internal/pkg/agent/application/monitoring/processes_cloud.go

```diff
@@ -18,7 +18,7 @@ func cloudComponentIDToAgentInputType(componentID string) string {
 	return componentID
 }
 
-func expectedCloudProcessID(c *component.Component) string {
+func expectedCloudProcessID(c component.Component) string {
```

Comment:
do we know what the impact (with particular regard to memory churn) is of copying the struct instead of just passing the pointer? Especially considering that the liveness  probe will run every few seconds ?

Reply:
So, I initially did that because it set off a linter error in `processesHandler()` 's `for` loop, as we're borrowing a reference from the for loop. This behavior is [changing in 1.22](https://go.dev/blog/loopvar-preview). I wasn't thinking about memory usage, though. I wonder if we can just only pass the parts of the component struct that we actually need.